--- a/src/modules/module-protocol-pulse/pulse-server.c
+++ b/src/modules/module-protocol-pulse/pulse-server.c
@@ -3046,10 +3046,15 @@
 	    (index != SPA_ID_INVALID && name != NULL))
 		return -EINVAL;
 
-	if (command == COMMAND_SET_SINK_MUTE)
+	if (command == COMMAND_SET_SINK_MUTE) {
+		if (client->quirks & QUIRK_BLOCK_SINK_VOLUME)
+			return -EPERM;
 		direction = PW_DIRECTION_OUTPUT;
-	else
+	} else {
+		if (client->quirks & QUIRK_BLOCK_SOURCE_VOLUME)
+			return -EPERM;
 		direction = PW_DIRECTION_INPUT;
+	}
 
 	o = find_device(client, index, name, direction == PW_DIRECTION_OUTPUT, &is_monitor);
 	if (o == NULL || (info = o->info) == NULL || info->props == NULL)
